src/lbaf/Execution/lbsPhaseStepperAlgorithm.py

```diff
@@ -0,0 +1,56 @@
+import sys
+import math
+import itertools
+from logging import Logger
+
+from .lbsAlgorithmBase import AlgorithmBase
+from ..Model.lbsObjectCommunicator import ObjectCommunicator
+from ..Model.lbsPhase import Phase
+from ..IO.lbsStatistics import print_function_statistics
+
+
+class PhaseStepperAlgorithm(AlgorithmBase):
+    """ A concrete class for the phase stepper non-optimzing algorithm
+    """
+
+    def __init__(self, work_model, parameters, lgr: Logger):
+        """ Class constructor
+            work_model: a WorkModelBase instance
+            parameters: a dictionary of parameters
+        """
+        # Call superclass init
+        super(PhaseStepperAlgorithm, self).__init__(work_model, parameters)
+
+        # Assign logger to instance variable
+        self.__logger = lgr
+
+    def execute(self, phases: list, distributions: dict, statistics: dict, _):
+        """ Execute brute force optimization algorithm on Phase instance
+        """
+        # Ensure that a list with at least one phase was provided
+        if not phases or not isinstance(phases, list) or not all(
+            [isinstance(p, Phase) for p in phases]):
```

Comment:
should be in line 31. Then line 31 will have 106 chars < 120, so it's ok, more readable

Reply:
I like better to see all conditions at a glance on a laptop

Reply:
Then it is again the same problem. Indentation is wrong, why? 
PEP 8: E125 continuation line with same indent as next logical line
Because the indentation should be different than lines inside if:
WRONG:
```Python3
if not phases or not isinstance(phases, list) or not all(
    [isinstance(p, Phase) for p in phases]):
    self.__logger.error(f"Algorithm execution requires a Phase instance")
    raise SystemExit(1)
```
CORRECT:
```Python3
if not phases or not isinstance(phases, list) or not all(
        [isinstance(p, Phase) for p in phases]):
    self.__logger.error(f"Algorithm execution requires a Phase instance")
    raise SystemExit(1)
```